services/streammanager/kafka/kafkamanager.go

```diff
@@ -450,14 +451,21 @@ func serializeAvroMessage(value []byte, codec goavro.Codec) ([]byte, error) {
 	return binary, nil
 }
 
-func prepareBatchOfMessages(topic string, batch []map[string]interface{}, timestamp time.Time, p producerManager) (
+func prepareBatchOfMessages(batch []map[string]interface{}, timestamp time.Time, p producerManager) (
 	[]client.Message, error,
 ) {
 	start := now()
 	defer func() { kafkaStats.prepareBatchTime.SendTiming(since(start)) }()
-
 	var messages []client.Message
+
 	for i, data := range batch {
+		topic := data["topic"].(string)
+
+		// we are guranatee presence of topic in each payload this is a fall back handler
+		if topic == "" {
+			return nil, fmt.Errorf("topic is empty for message %d", i)
```

Comment:
please make sure that this error is treated at 400 abortable error

Reply:
Yes this will be 400 outside where its handled

Reply:
ref: https://play.golang.com/p/YTTFP5ZSc_C
What about following?
```go
topic, ok := data["topic"].(string)
if !ok || topic == "" {
  return nil, fmt.Errorf("topic is invalid for message %d", i)
}
```

Reply:
Error message can be extended to "topic is invalid for the event, it should be non-empty string %d"